ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/avdtestbase.py

```diff
@@ -43,11 +46,37 @@ def render(self) -> dict:
         """
         return getattr(self, "test_definition", None) or {}
 
+    def get_host_structured_config(self, host: str) -> dict:
+        """
+        Retrieves and returns the structured configuration for a specified host.
+
+        The function fetches the structured_config from hostvars. It ensures the returned object is a
+        dictionary, converting Ansible dictionary-like objects to a standard dictionary if necessary.
+
+        Args:
+            host (str): Hostname to retrieve the structured_config.
+
+        Returns:
+            dict: Structured configuration for the host.
+        """
+        if host not in self.hostvars:
+            raise AristaAvdError(f"Host '{host}' is missing from the hostvars.")
+        struct_cfg = get(self.hostvars, host, separator="..")
```

Comment:
Since you check just above that `hostvars`` has the key `host` you can just use `struct_cfg = self.hostvars[host]`